core/src/test/java/kafka/server/share/SharePartitionManagerTest.java

```diff
@@ -1221,6 +1224,36 @@ public void testReplicaManagerFetchShouldProceed() {
             any(), any(), any(ReplicaQuota.class), any());
     }
 
+    @Test
+    public void testCloseSharePartitionManager() throws Exception {
+        SharePartitionManager sharePartitionManager = SharePartitionManagerBuilder.builder()
+                .withShareGroupPersister(mock(Persister.class)).build();
+
+        List<Integer> mockList = new ArrayList<>();
+        sharePartitionManager.timer().add(createTimerTask(mockList));
+        // Allowing first timer task to expire. The timer task will add an element to mockList.
+        TestUtils.waitForCondition(
+                () -> mockList.size() == 1,
+                DEFAULT_MAX_WAIT_TIMER_TASK_TIMEOUT_MS,
+                () -> "Timer task never got timed out.");
+
+        sharePartitionManager.timer().add(createTimerTask(mockList));
+        // Closing the sharePartitionManager closes timer object in sharePartitionManager.
+        sharePartitionManager.close();
+        // Allowing second timer task to expire. It won't be able to add an element to mockList.
+        Thread.sleep(DEFAULT_MAX_WAIT_TIMER_TASK_TIMEOUT_MS);
```

Comment:
I don't think it's a good idea to sleep. Can't we use mocktimer?

Reply:
For that, we need to have a function in SharePartitionManager to set the timer object to this mockTimer object. We can mark it as `// Only for testing.` Do we want to take that route?

Reply:
Also mockTimer does not support close() functionality. It just throws an exception as indicated [here](https://github.com/apache/kafka/blob/trunk/server-common/src/test/java/org/apache/kafka/server/util/timer/MockTimer.java#L92).

Reply:
my bad, we can just use `SystemTimerReaper` class and use the `advanceClock` functionality from that class.

Reply:
We have used `Mockito.verify` to verify the number of `close()` calls to mock timer object.